MINOR: Replace TopicAndPartition with TopicPartition in `Log` and `ReplicaManager`

ijuma opened the pull request.

ijuma committed:
MINOR: Replace TopicAndPartition with TopicPartition in `Log` and `ReplicaManager`

And other code that calls those two classes. To be done separately: Controller and
ZkUtils.

ijuma commented:
retest this please

ijuma commented:
Review by @hachikuji.

ijuma committed:
Remove unnecessary allocation of `TopicPartition`

hachikuji commented on core/src/main/scala/kafka/log/LogCleaner.scala:
```diff
@@ -127,7 +128,7 @@ class LogCleaner(val config: CleanerConfig,
    *  the partition is aborted.
    */
-  def abortCleaning(topicAndPartition: TopicAndPartition) {
+  def abortCleaning(topicAndPartition: TopicPartition) {
```
In other places, we renamed `topicAndPartition` to `topicPartition`. Guess we could do the same here for consistency? A few more of these below.

hachikuji commented on core/src/main/scala/kafka/log/LogCleaner.scala:
```diff
@@ -172,7 +173,7 @@ class LogCleaner(val config: CleanerConfig,
    * @return A boolean indicating whether the work has completed before timeout
    */
   def awaitCleaned(topic: String, part: Int, offset: Long, maxWaitMs: Long = 60000L): Boolean = {
```
Would it be more consistent to let this accept `TopicPartition`?

hachikuji commented on core/src/main/scala/kafka/log/LogCleanerManager.scala:
```diff
@@ -22,11 +22,12 @@ import java.util.concurrent.TimeUnit
 
 import com.yammer.metrics.core.Gauge
-import kafka.common.{LogCleaningAbortedException, TopicAndPartition}
+import kafka.common.{LogCleaningAbortedException}
```
nit: the braces seem unneeded.

hachikuji commented on core/src/main/scala/kafka/log/LogManager.scala:
```diff
@@ -287,7 +289,7 @@ class LogManager(val logDirs: Array[File],
    * @param partitionAndOffsets Partition logs that need to be truncated
    */
-  def truncateTo(partitionAndOffsets: Map[TopicAndPartition, Long]) {
+  def truncateTo(partitionAndOffsets: Map[TopicPartition, Long]) {
```
nit: maybe we should remove the `And` from `partitionAndOffsets`?

hachikuji commented on core/src/main/scala/kafka/server/ReplicaFetcherThread.scala:
```diff
@@ -24,7 +24,7 @@ import kafka.admin.AdminUtils
 import kafka.log.LogConfig
 import kafka.api.{KAFKA_0_10_0_IV0, KAFKA_0_10_1_IV1, KAFKA_0_10_1_IV2, KAFKA_0_9_0}
-import kafka.common.{KafkaStorageException, TopicAndPartition}
+import kafka.common.{KafkaStorageException}
```
nit: unneeded braces

hachikuji commented on core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala:
```diff
@@ -102,8 +102,8 @@ class LogCleanerManagerTest extends JUnitSuite with Logging {
-    val topicAndPartition = TopicAndPartition("log", 0)
-    val lastClean = Map(topicAndPartition-> 0L)
+    val topicAndPartition = new TopicPartition("log", 0)
+    val lastClean = Map(topicAndPartition -> 0L)
```
Any reason not to fix the other couple lines like this below?

hachikuji commented on core/src/test/scala/unit/kafka/server/LogOffsetTest.scala:
```diff
@@ -248,9 +248,6 @@ class LogOffsetTest extends ZooKeeperTestHarness {
-    val dir = TestUtils.tempDir()
-    dir
-  }
+  private def getLogDir(): File = TestUtils.tempDir()
```
nit: this function doesn't contribute much.

hachikuji reviewed (commented):
Thanks for the patch. Looks good, but left a few nitpicks to keep up appearances.

ijuma committed:
Rename topicAndPartition to topicPartition in a number of places

ijuma committed:
Remove `getLogDir` as it doesn’t provide much over `TestUtils.tempDir`

ijuma commented:
Thanks for the review @hachikuji. I believe I have addressed your comments. I had initially not done all the `topicAndPartition` to `topicPartition` renames to avoid making the diff too large. However, inspired by your review, I went ahead and tried to do a more thorough job.

ijuma commented:
I rebased to include #2280 to ensure that there are no merge conflicts. #2280 should be merged first anyway.

ijuma commented:
Started system tests build:

https://jenkins.confluent.io/job/system-test-kafka-branch-builder/636/

ijuma commented:
retest this please

hachikuji commented on core/src/main/scala/kafka/server/ReplicaFetcherThread.scala:
```diff
@@ -139,7 +139,7 @@ class ReplicaFetcherThread(name: String,
         trace("Follower %d set replica high watermark for partition [%s,%d] to %s"
           .format(replica.brokerId, topic, partitionId, followerHighWatermark))
-      if (quota.isThrottled(TopicAndPartition(topic, partitionId)))
+      if (quota.isThrottled(new TopicPartition(topic, partitionId)))
```
Seems we can just use `topicPartition`?

hachikuji commented on core/src/main/scala/kafka/server/ReplicaManager.scala:
```diff
@@ -621,15 +616,15 @@ class ReplicaManager(val config: KafkaConfig,
-  def getMagicAndTimestampType(topicAndPartition: TopicAndPartition): Option[(Byte, TimestampType)] =
-    getReplica(topicAndPartition.topic, topicAndPartition.partition).flatMap { replica =>
+  def getMagicAndTimestampType(topicPartition: TopicPartition): Option[(Byte, TimestampType)] =
+    getReplica(topicPartition.topic, topicPartition.partition).flatMap { replica =>
```
It's a little weird that the `allPartitions` map in this class is keyed by a tuple instead of `TopicPartition`. Think it's worth changing here?

ijuma committed:
Pass `topicPartition` instead of allocating a new one in `ReplicaFetcherThread`

Also include minor clean-up in `ReplicaManager.getReplica`. I considered changing
it to accept `TopicPartition`, but a lot of other methods in that class take topic and
partition as separate parameters.

ijuma committed:
Use `TopicPartition` instead of tuple for `allPartitions` map

ijuma commented:
@hachikuji, addressed both comments.

hachikuji commented on core/src/main/scala/kafka/server/ReplicaManager.scala:
```diff
@@ -269,20 +268,16 @@ class ReplicaManager(val config: KafkaConfig,
-  def getPartition(topic: String, partitionId: Int): Option[Partition] = {
-    Option(allPartitions.get((topic, partitionId)))
-  }
+  def getPartition(topic: String, partitionId: Int): Option[Partition] =
```
If you're keen to go one more step, we could let these methods accept `TopicPartition` also.

ijuma committed:
Change `ReplicaManager` methods to take `TopicPartition` instead of `topic` and `partition` separately.

hachikuji commented:
@ijuma LGTM. Nice piece of cleanup!

ijuma committed:
Remove unnecessary/unused vals for `topic` and `partition`

ijuma committed:
Fix initialisation order bug in `LogRecoveryTest`
